polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp

```diff
@@ -1230,54 +1286,38 @@ void MemoryAccessAnalysis::build(T memoryOp, DataFlowSolver &solver) {
     llvm::dbgs() << "\n";
   });
 
-  // Collect the "get_global_ids" operations (yielding the global thread ids).
-  std::vector<sycl::SYCLNDItemGetGlobalIDOp> getGlobalIdOps =
-      getOperationsOfType<sycl::SYCLNDItemGetGlobalIDOp>(
-          memoryOp->template getParentOfType<FunctionOpInterface>())
-          .takeVector();
-
-  // Return the index value of a "get_global_ids" operation.
-  auto getIndexValue =
-      [&](sycl::SYCLNDItemGetGlobalIDOp &op) -> std::optional<APInt> {
-    std::optional<TypedValue<IntegerType>> idx = op.getIndex();
-    return idx.has_value() ? getConstIntegerValue(*idx, solver) : APInt();
-  };
-
-  // Ensure that all "get_global_ids" have index values known at compile time.
-  if (llvm::any_of(getGlobalIdOps, [&](sycl::SYCLNDItemGetGlobalIDOp &op) {
-        return !getIndexValue(op).has_value();
-      })) {
-    LLVM_DEBUG(llvm::dbgs() << "Cannot order 'get_global_ids' operations in "
-                               "increasing index value.\n");
-    return;
-  }
-
-  // Order the "get_global_ids" operations in increasing index value.
-  llvm::sort(getGlobalIdOps, [&](sycl::SYCLNDItemGetGlobalIDOp &op1,
-                                 sycl::SYCLNDItemGetGlobalIDOp &op2) {
-    return getIndexValue(op1)->slt(*getIndexValue(op2));
-  });
+  // Collect the global thread ids used in the function.
+  auto funcOp = memoryOp->template getParentOfType<FunctionOpInterface>();
+  SmallVector<Value> threadVars = getThreadVector(funcOp, solver);
```

Comment:
Would it make sense to add an early return here, in case `threadVars` is empty (i.e., we didn't find `nd_item` or `item` or we found both)?

Reply:
Ok, makes sense 👍